src/disp_s1/ionosphere.py

```diff
@@ -303,17 +305,15 @@ def download_ionosphere_files(config: DownloadConfig) -> list[Path]:
     session = SessionWithHeaderRedirection(username, password)
     downloaded_files = []
 
-    for input_file in config.input_files:
-        # Try parsing as SAFE first, then CSLC
-        try:
-            date_str = parse_safe_date(input_file)
-        except ValueError:
-            try:
-                date_str = parse_cslc_date(input_file)
-            except ValueError:
-                raise ValueError(
-                    f"File {input_file} does not match SAFE or CSLC naming convention"
-                )
+    date_list = []
+    for file in config.input_files:
+        date_list.extend(get_dates(file))
+
+    date_list = [
+        datetime.datetime.strftime(date, "%Y%m%d") for date in list(set(date_list))
+    ]
+
+    for date_str in date_list:
 
         archive_date = ArchiveDate.from_date_str(date_str)
```

Comment:
I'm also noticing that this `ArchiveDate` is kinda weird in that we have to go from string -> `datetime` using `get_dates` -> string (using `strftime`) just so we can call `from_date_str`.

The `from_date_str` should probably be `from_datetime`

Reply:
yes I agree with you, both of your comments suggest me to update the compressed SLC naming instead is a better choice